diff --git a/meson.build b/meson.build
index c1ea053..7bd7380 100644
--- a/meson.build
+++ b/meson.build
@@ -102,7 +102,6 @@ glib_dep = dependency('glib-2.0', version: '>= 2.66')
 gio_dep = dependency('gio-2.0')
 gio_unix_dep = dependency('gio-unix-2.0')
 json_glib_dep = dependency('json-glib-1.0')
-fuse3_dep = dependency('fuse3', version: '>= 3.10.0')
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0')
 geoclue_dep = dependency('libgeoclue-2.0',
                          version: '>= 2.5.2',
@@ -184,8 +183,6 @@ pkgconfig.generate(
 
 subdir('data')
 subdir('src')
-subdir('document-portal')
-subdir('tests')
 subdir('po')
 subdir('doc')
 
@@ -193,12 +190,9 @@ subdir('doc')
 configure_file(output: 'config.h', configuration: config_h)
 
 summary({
-    'Enable docbook documentation': build_docbook,
     'Enable libsystemd support': have_libsystemd,
     'Enable geoclue support': have_geoclue,
     'Enable libportal support': have_libportal,
-    'Enable installed tests:': enable_installed_tests,
-    'Enable python test suite': enable_pytest,
     'Build man pages': rst2man.found(),
     'Build flatpak interfaces': flatpak_intf_dir != '',
     'Sandboxed image validation': use_bwrap,
